templates/e2e-dspy-workflow/README.ipynb

```diff
@@ -0,0 +1,1074 @@
+    "We also pass in the valid labels to the module.\n",
+    "\n",
+    "Inside the forward method, we pass the text to the predictor, do a little cleaning, and return the prediction."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 13,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "class IntentClassificationModule(dspy.Module):\n",
+      "    def __init__(self, labels_in_use):\n",
+      "        self.intent_classifier = dspy.ChainOfThought(IntentClassification)\n",
+      "        self.valid_labels = set(labels_in_use)\n",
+      "\n",
+      "    def forward(self, text):\n",
+      "        prediction = self.intent_classifier(intent=text)\n",
+      "        sanitized_prediction = dspy.Prediction(label=prediction.label.lower().strip().replace(\" \", \"_\"), reasoning=prediction.reasoning)\n",
+      "        return sanitized_prediction\n",
+      "\n"
+     ]
+    }
+   ],
+   "source": [
+    "print(inspect.getsource(IntentClassificationModule))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Lastly, we set up some the vanilla program we will use throughout the notebook."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 14,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from src import MODEL_PARAMETERS, LOCAL_API_PARAMETERS\n",
+    "vanilla_program = IntentClassificationModule(labels_in_use)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 15,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Note: Run above this to do all setup without launching any models\n",
+    "# This is useful if you have already collected data and want to start from finetuning or from evaluation"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We will be using a local VLLM instance to run the initial benchmarks and data collection."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Gathering training data and running the 70B Model\n",
+    "\n",
+    "\n",
+    "## Preparation\n",
+    "\n",
+    "Before running the 70B model:\n",
+    "1. Remember to set your HF_TOKEN and HF_HOME environment variables\n",
+    "2. Use the following command to start the 70B server:\n",
+    "\n",
+    "   ```\n",
+    "   vllm serve meta-llama/Meta-Llama-3.1-70B-Instruct --port 8000 --pipeline_parallel_size 2 --enable_prefix_caching --tensor_parallel_size 2\n",
+    "   ```\n",
+    "\n",
+    "## Parallelism Configuration\n",
+    "\n",
+    "We've chosen pipeline parallelism and tensor parallelism of 2 for the 70B model based on our current setup. Here's the reasoning:\n",
+    "\n",
+    "1. Model size: The 70B model has 30 parts of ~5 GB each (based on [HuggingFace documentation](https://huggingface.co/meta-llama/Meta-Llama-3.1-70B-Instruct/tree/main)).\n",
```

Comment:
Hmm this can even be removed since it's not a focus for the notebook. Or if oyu really want to keep it make it a dropdown widget (like our ray summit notebooks)